### PR TITLE
Added proxy sample

### LINKED ISSUE #78: No example of how to setup client using proxy
1. Please check our current Issues to see if someone already reported this https://github.com/Microsoft/typed-rest-client/issues
2. Take a look at our Instructions for logging issues https://github.com/Microsoft/typed-rest-client/blob/master/CONTRIBUTING.md#instructions-for-logging-issues

### Environment
Node version: 8.9.4
Npm version: 5.6.0
OS and version: MacOS High Sierra
typed-rest-client version: 1.0.9

### Issue Description
I can't configure the proxy. Not sure if there is a bug or if I messed up sth.
Also, my script is run by protractor with a configured proxy

### Expected behaviour
Add example how to use a proxy (and verify it)

### Actual behaviour
I got the following error: 
tunneling socket could not be established, cause=connect ETIMEDOUT ip address:port

### Steps to reproduce
1. setup proxy
2. try to configure HTTPClient to work with this proxy

### PR DESCRIPTION
**Pull request by damccorm**
Resolves #78

### CODE REVIEW

File: samples/http.ts [hunk 1]
```diff
@@ -28,6 +28,25 @@ export async function run() {
...
+        // Http get request using a proxy
+        //
+        cm.heading('get request using the proxy url set in the env variables');
+        let proxySettings = {
+            proxyUrl: cm.getEnv('PROXY_URL'),
+            proxyUsername: cm.getEnv('PROXY_USERNAME'),
+            proxyPassword: cm.getEnv('PROXY_PASSWORD'),
+            proxyBypassHosts: cm.getEnv('PROXY_BYPASS_HOSTS') ? cm.getEnv('PROXY_BYPASS_HOSTS').split(', ') : null
+        }
+        if (proxySettings.proxyUrl){
```
Comment (stephenmichaelf) [inline samples/http.ts]: Nit: space before brace.

File: samples/http.ts [hunk 2]
```diff
@@ -28,6 +28,25 @@ export async function run() {
...
+            httpc = new httpm.HttpClient('vsts-node-api', null, { proxy: proxySettings });
+            body = await (await httpc.get('https://httpbin.org/get')).readBody();
+            cm.outputHttpBinResponse(body);
+        }
+        else{
```
Comment (stephenmichaelf) [inline samples/http.ts]: Nit: space before brace.

**Comment by stephenmichaelf** [comment]
@zielinskikamil Does this proxy example help you with your use case?

**Comment by damccorm** [comment]
@zielinskikamil, any update? Did this help?

**Comment by zielinskikamil** [comment]
@stephenmichaelf @damccorm Thanks for that. It helped me a lot!

**Pull request merged**